addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/NetworkOceanicThingHandler.java

```diff
@@ -0,0 +1,221 @@
+                    config.portNumber, e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Could not connect to host " + config.ipAddress + ": " + e.getMessage());
+        }
+    }
+
+    @Override
+    public void dispose() {
+        NetworkOceanicBindingConfiguration config = getConfigAs(NetworkOceanicBindingConfiguration.class);
+
+        IOUtils.closeQuietly(inputStream);
+        IOUtils.closeQuietly(outputStream);
+        if (socket != null) {
+            try {
+                socket.close();
+                socket = null;
+            } catch (IOException e) {
+                logger.error("An exception ocurred while disconnecting to host {}:{} : '{}'", config.ipAddress,
+                        config.portNumber, e.getMessage());
+            }
+        }
+
+        super.dispose();
+    }
+
+    @Override
+    protected String requestResponse(String commandAsString) {
+        synchronized (this) {
+            if (getThing().getStatus() == ThingStatus.ONLINE) {
+                NetworkOceanicBindingConfiguration config = getConfigAs(NetworkOceanicBindingConfiguration.class);
+                Throttler.lock(config.ipAddress);
+
+                String request = commandAsString + "\r";
+
+                byte[] dataBuffer = new byte[bufferSize];
+                byte[] tmpData = new byte[bufferSize];
+                String line;
+                int len = -1;
+                int index = 0;
+                boolean sequenceFound = false;
+
+                final byte lineFeed = (byte) '\n';
+                final byte carriageReturn = (byte) '\r';
+                final byte nullChar = (byte) '\0';
+                final byte eChar = (byte) 'E';
+                final byte rChar = (byte) 'R';
+
+                try {
+                    logger.debug("Sending request '{}'", request);
+
+                    outputStream.write(request.getBytes());
+                    outputStream.flush();
+
+                    while (true) {
+                        if ((len = inputStream.read(tmpData)) > -1) {
+                            if (logger.isTraceEnabled()) {
+                                StringBuilder sb = new StringBuilder();
+                                for (int i = 0; i < len; i++) {
+                                    sb.append(String.format("%02X", tmpData[i]));
+                                    sb.append(" ");
+                                }
+                                logger.trace("Read {} bytes : {}", len, sb.toString());
+                            }
+
+                            for (int i = 0; i < len; i++) {
+                                if (logger.isTraceEnabled()) {
+                                    logger.trace("Byte {} equals '{}' (hex '{}')", i,
+                                            new String(new byte[] { tmpData[i] }), String.format("%02X", tmpData[i]));
```

Comment:
I prefer to have in in HEX

Reply:
It does the same as `new String(new byte[] { tmpData[i] })`, so it would become:

```java
logger.trace("Byte {} equals '{}' (hex '{}')", i, 
String.format("%c", tmpData[i]), String.format("%02X", tmpData[i]));
```

Reply:
@kgoderis you still did not seem to have replied to this comment?

Reply:
But I meant this one ☺️